### ISSUE TITLE
Bug in conversion frompd.series in 0.15.1

### ISSUE THREAD

**Issue by kefirbandi** [pydata/xarray#4087]
Bug in conversion frompd.series 

#### MCVE Code Sample

```python
import pandas as pd
import xarray as xr

data3 = pd.DataFrame([(1,  2, 1), (2, 2, 1)],columns=['x', 'B', 'A'])
data3 = data3.set_index(['x'])
data3.rename_axis('tag', axis=1, inplace=True)
data3 = data3.stack()
print(data3)
print(xr.DataArray.from_series(data3).sel(tag='B'))
```

#### Expected Output
[2,2]

#### Problem Description
The same code gives the expected output in xarray 0.14.1
Also works correctly if "tag" dimension is properly sorted during DataFrame cretion.

#### Versions

<details><summary>Output of <tt>xr.show_versions()</tt></summary>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.7 | packaged by conda-forge | (default, Jul  2 2019, 02:18:42) 
[GCC 7.3.0]
python-bits: 64
OS: Linux
OS-release: 4.1.12-124.38.1.el7uek.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: en_US.utf8
LANG: en_US.utf8
LOCALE: en_US.UTF-8
libhdf5: 1.10.4
libnetcdf: 4.6.3

xarray: 0.15.1
pandas: 0.25.0
numpy: 1.17.0
scipy: 1.3.0
netCDF4: 1.5.3
pydap: None
h5netcdf: None
h5py: 2.7.1
Nio: None
zarr: None
cftime: 1.1.3
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: None
dask: 2.16.0
distributed: None
matplotlib: 3.1.1
cartopy: None
seaborn: 0.9.0
numbagg: None
setuptools: 46.3.1
pip: 19.2.1
conda: 4.8.3
pytest: 5.3.0
IPython: 7.7.0
sphinx: 2.1.2

</details>

**Comment by keewis** [comment]
that's strange. If I run it, I get
```python
In [1]: import pandas as pd 
   ...: import xarray as xr

In [2]: data3 = pd.DataFrame([(1,  2, 1), (2, 2, 1)],columns=['x', 'B', 'A']) 
   ...: data3 = data3.set_index(['x']) 
   ...: data3.rename_axis('tag', axis=1, inplace=True) 
   ...: data3 = data3.stack() 
   ...:  
   ...: data3
Out[2]: 
x  tag
1  B      2
   A      1
2  B      2
   A      1
dtype: int64

In [3]: xr.DataArray.from_series(data3).sel(tag='B')
Out[3]: 
<xarray.DataArray (x: 2)>
array([2, 2])
Coordinates:
  * x        (x) int64 1 2
    tag      <U1 'B'
```
which I believe is what you expect.

**Comment by kmuehlbauer** [comment]
@keewis I've the same output as the OP. Here is my xr.show_versions():

```
INSTALLED VERSIONS
------------------
commit: None
python: 3.8.2 | packaged by conda-forge | (default, Apr 16 2020, 18:04:51) 
[GCC 7.3.0]
python-bits: 64
OS: Linux
OS-release: 4.12.14-lp151.28.48-default
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: de_DE.UTF-8
LOCALE: de_DE.UTF-8
libhdf5: 1.10.5
libnetcdf: 4.7.4

xarray: 0.15.1
pandas: 1.0.3
numpy: 1.18.1
scipy: 1.4.1
netCDF4: 1.5.3
pydap: None
h5netcdf: 0.8.0
h5py: 2.10.0
Nio: None
zarr: None
cftime: 1.1.1.2
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: 1.3.2
dask: 2.14.0
distributed: 2.14.0
matplotlib: 3.2.1
cartopy: 0.17.0
seaborn: None
numbagg: None
setuptools: 46.1.3.post20200325
pip: 20.0.2
conda: None
pytest: 5.4.1
IPython: 7.13.0
sphinx: None
```

**Comment by keewis** [comment]
thanks for the hint. Yes, that seems to have been fixed since the last release (#3951, I believe). It's going to be part of the next release, which should be quite soon.
